roleutils/buttonroles.py

```diff
@@ -0,0 +1,441 @@
+"""
+MIT License
+
+Copyright (c) 2020-2023 PhenoM4n4n
+Copyright (c) 2023-present japandotorg
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import asyncio
+import discord
+import logging
+from typing import Any, Dict, Optional, Union
+
+from redbot.core import commands, Config
+from redbot.core.bot import Red
+from redbot.core.utils import menus
+from redbot.core.utils.predicates import MessagePredicate
+from redbot.core.utils.chat_formatting import pagify
+
+from .abc import MixinMeta, CompositeMetaClass
+from .converters import EmojiRole
+
+log = logging.getLogger("red.roleutils.buttonroles")
+
+
+class ButtonRoles(MixinMeta, metaclass=CompositeMetaClass):
+    """Manage button-based role assignment."""
+
+    def __init__(self, *_args: Any) -> None:
+        super().__init__(*_args)
+        self.method: str = "build"
```

Comment:
~~no need to use this here, remove~~ (canceling this one rn, you're somehow making use of it, we'll try to adjust that later)